internal/provider/resource_access_model.go

```diff
@@ -417,3 +428,62 @@ func retrieveAccessModelDataFromID(ctx context.Context, ID types.String, users t
 
 	return resID[0], resID[1], stateUsers
 }
+
+// UpgradeState upgrades the state of the access model resource.
+// This is used to handle changes in the resource schema between versions.
+func (o *accessModelResource) UpgradeState(ctx context.Context) map[int64]resource.StateUpgrader {
+	return map[int64]resource.StateUpgrader{
+		// Upgrade from list to set for `users`.
+		0: {
+			PriorSchema: &schema.Schema{
+				Attributes: map[string]schema.Attribute{
```

Comment:
When you merge this forward to v1 I suggest following the style there of moving the old schema into a separate function.